common/rya.api/src/main/java/org/apache/rya/api/RdfCloudTripleStoreConfigurationBuilder.java

```diff
@@ -0,0 +1,158 @@
+     * 
+     * @return specified builder
+     */
+    protected abstract B confBuilder();
+
+    /**
+     * Auxiliary method to return type of configuration object constructed by
+     * builder in a type safe way
+     * 
+     * @return specified configuration
+     */
+    protected abstract C createConf();
+
+    /**
+     * Set whether to use backwards chaining inferencing during query
+     * evaluation. The default value is false.
+     * 
+     * @param useInference
+     *            - turn inferencing on and off in Rya
+     * @return B - concrete builder class
+     */
+    public B useInference(boolean useInference) {
+        this.useInference = useInference;
+        return confBuilder();
+    }
+
+    /**
+     * 
+     * Sets the authorization for querying the underlying data store.
+     * 
+     * @param auths
+     *            - authorizations for querying underlying datastore
+     * @return B - concrete builder class
+     */
+    public B auths(String auths) {
+        Preconditions.checkNotNull(auths, "User auths cannot be null.");
+        this.auths = auths;
+        return confBuilder();
+    }
+
+    /**
+     * Set the column visibities for ingested triples. If no value is set,
+     * triples won't have a visibility.
+     * 
+     * @param visibilities
+     *            - visibilities assigned to any triples inserted into Rya
+     * @return B - concrete builder class
+     */
+    public B visibilities(String visibilites) {
+        Preconditions.checkNotNull(visibilites, "Visibilities cannot be null.");
+        this.visibilities = visibilites;
+        return confBuilder();
+    }
+
+    /**
+     *
+     * Sets the prefix for the Rya instance to connect to. This parameter is
+     * required and the default value is "rya_"
+     * 
+     * @param prefix
+     *            - the prefix for the Rya instance
+     * @return B - concrete builder class
+     */
+    public B setRyaPrefix(String prefix) {
+        Preconditions.checkNotNull(prefix, "Rya prefix cannot be null.");
+        this.prefix = prefix;
+        return confBuilder();
+    }
+
+    /**
+     * Set whether to display query plan during optimization. The default value
+     * is false.
+     * 
+     * @param displayPlan
+     *            - display the parsed query plan during query evaluation
+     *            (useful for debugging)
+     * @return B - concrete builder class
+     */
+    public B displayQueryPlan(boolean displayPlan) {
+        this.displayPlan = displayPlan;
+        return confBuilder();
+    }
+
+    public C build() {
+        return getConf(createConf());
+    }
+
+    private C getConf(C conf) {
+
+        Preconditions.checkNotNull(auths, "Authorizations cannot be null");
+
+        conf.setInfer(useInference);
+        conf.setTablePrefix(prefix);
+        conf.setInt("sc.cloudbase.numPartitions", 3);
+        conf.set(RdfCloudTripleStoreConfiguration.CONF_QUERY_AUTH, auths);
```

Comment:
RdfCloudTripleStoreConfiguration shoudl implement a method for setting query auths. Within that method, do the preconditions check instead of within this builder.